machine-learning/02-Data Preparation and Feature Engineering.ipynb

```diff
@@ -1,45 +1,405 @@
+      "source": [
+        "### What is an IDataView?\n",
+        "\n",
+        "The [IDataView](https://docs.microsoft.com/dotnet/api/microsoft.ml.idataview?view=ml-dotnet) is the data format ML.NET loads for training. It is a set of interfaces and components that provide efficient, compositional processing of schematized data for machine learning and advanced analytics applications. It is designed to gracefully and efficiently handle high dimensional data and large data sets. \n",
+        "\n",
+        "The IDataView has general schema support, in that a view can have an arbitrary number of columns, each having an associated name, index, data type, and optional annotation."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### How to create an IDataView\n",
+        "\n",
+        "You can create an IDataView by using any of the methods for loading data:\n",
+        "\n",
+        "- TextLoader\n",
+        "- LoadFromEnumerable\n",
+        "- DatabaseLoader\n",
+        "- LoadFromTextFile\n",
+        "\n",
+        "See [Load Data from files and other sources](https://docs.microsoft.com/en-us/dotnet/machine-learning/how-to-guides/load-data-ml-net) for further documentation and examples. "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "dotnet_interactive": {
+          "language": "csharp"
+        },
+        "vscode": {
+          "languageId": "dotnet-interactive.csharp"
+        }
+      },
+      "outputs": [
+        {
+          "data": {
+            "text/html": [
+              "<div><div></div><div></div><div><strong>Installed Packages</strong><ul><li><span>Microsoft.ML, 1.7.1</span></li></ul></div></div>"
+            ]
+          },
+          "metadata": {},
+          "output_type": "display_data"
+        }
+      ],
+      "source": [
+        "#r \"nuget: Microsoft.ML, 1.7.1\""
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "dotnet_interactive": {
+          "language": "csharp"
+        },
+        "vscode": {
+          "languageId": "dotnet-interactive.csharp"
+        }
+      },
+      "outputs": [],
+      "source": [
+        "using Microsoft.ML;\n",
+        "using Microsoft.ML.Data;"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "#### Loading from a file\n",
+        "\n",
+        "A [TextLoader](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.data.textloader?view=ml-dotnet) can load a structred file into the IDataView format. Structured information is basically columns and rows of data. The loader will need to be told if the file has a header, and what separator character the file uses. \n",
+        "\n",
+        "By using the loader instead of the direct method LoadFromTextFile, there is the option to load from multiple files. "
```

Comment:
I'm not sure this is completely accurate. You can use `LoadFromTextFile` to load from [multiple files that are in the same directory](https://docs.microsoft.com/dotnet/machine-learning/how-to-guides/load-data-ml-net#load-from-files-in-a-single-directory) using wildcards. What you can do with `TextLoader` though is load [multiple files in separate directories](https://docs.microsoft.com/dotnet/machine-learning/how-to-guides/load-data-ml-net#load-from-files-in-multiple-directories).

Reply:
Good point. I'll update to "load files from multiple locations".